rclcpp/src/rclcpp/node_interfaces/node_parameters.cpp

```diff
@@ -50,6 +51,15 @@ NodeParameters::NodeParameters(
     rmw_qos_profile_parameter_events,
     use_intra_process,
     allocator);
+
+  // TODO(sloretz) store initial values and use them when a parameter is created
```

Comment:
can you explain this TODO a bit more for my understanding please? The impression I get is that one day we want the node to store a list of possible initial parameter values, but not actually set those parameters until a different step?

What we are doing here -- which at the moment is essentially passing in a list of initial parameters -- is not the end goal? Why not?

I have seen in the parent ticket that this is "Needed to initialize a node with parameters generated by roslaunch after it merges multiple yaml files together." but it wasn't enough for me to answer my own question

Reply:
> The impression I get is that one day we want the node to store a list of possible initial parameter values, but not actually set those parameters until a different step?

Yeah, the TODO says what should happen here when (#475) is implemented. The reason for having a different step is to allow a node author to define parameters, including things like a `read_only` parameter that can be initialized but not modified. The parameter shouldn't be set until a node-author finishes defining parameters so that it's possible to error when a run-time user has a typo in their configuration.

> What we are doing here -- which at the moment is essentially passing in a list of initial parameters -- is not the end goal? Why not?

Passing in parameters via constructor will still exist, but setting them right away instead of storing them is the back-up plan. If the whole feature doesn't get in by bouncy release then this at least allows run-time users to initialize parameters via the command line.

Reply:
OK, 475 seems to be the missing piece in the explanation puzzle, thanks for explaining

Reply:
Referenced issue in TODO comment f30c64cb494351387813395c85e64f40ae3ee4ba